### ISSUE TITLE
AWS CloudFront HTTP/2 support

### ISSUE THREAD

**Issue by gips0n** [hashicorp/terraform#8730]
AWS [announced support](https://aws.amazon.com/about-aws/whats-new/2016/09/amazon-cloudfront-now-supports-http2/) for HTTP/2 protocol in Amazon CloudFront.

It would be nice to support configuration of `Supported HTTP Versions` in terraform

**Comment by abevoelker** [comment]
Yes this would be nice, because I just created a CloudFront distrubtion using the Terraform module and it defaulted to `HTTP/1.1, HTTP/1.0` rather than `HTTP/2, HTTP/1.1, HTTP/1.0`.

**Comment by jtdowney** [comment]
Now that aws/aws-sdk-go@381675cd10eb22e580b6e37fb433124592663783 has been committed, it shouldn't be too hard to add. If no one else submits a PR for it this weekend I might take a crack at it.

**Comment by vancluever** [comment]
Hey guys, working on this one now.

@abevoelker it's funny you mention that TF was defaulting to 1.1 though - the docs seem to state that cloudfront will set up HTTP2 by default:

```
// (Optional) Specify the maximum HTTP version that you want viewers to use
// to communicate with CloudFront. The default value for new web distributions
// is http2. Viewers that don't support HTTP/2 will automatically use an earlier
// version.
```

Anyway TF's default will explicitly be HTTP2 so you will get that regardless of what the API is giving you. It could be that the distribution was already created as well (not your fault, just a matter of circumstance).

**Comment by abevoelker** [comment]
@vancluever Hmm it's possible I'm doing something wrong, but I just created 2 more `aws_cloudfront_distribution` resources yesterday using TF and in both cases I had to login to AWS console and manually select HTTP/2 after the fact. Thanks for working on this!

**Comment by stack72** [comment]
Closed via #8777
